CMakeLists.txt

```diff
@@ -763,6 +763,31 @@ check_c_source_compiles("
 cmake_pop_check_state()
 tuklib_add_definition_if(liblzma HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR)
 
+# Check for __attribute__((ifunc())) support.
+option(USE_ATTR_IFUNC "Use __attribute__((ifunc())) if supported by the system"
+        ON)
+
+if (USE_ATTR_IFUNC)
```

Comment:
This option name is a bit misleading. Enabling it will only check for __attribute__((ifunc())) support, not guarantee they are used. Maybe rename it to CHECK_ATTR_IFUNC?